lib/api.js

```diff
@@ -66,221 +112,229 @@ var GeotabApi = function (getCredentialsCallback, newOptions, customCredentialSt
+              callbackError('Cancelled', {});
+            }
           }
         }
+      } else {
+        return JSONPromise;
       }
-    },
+    };
     /**
-     *  Construct an XMLHttpRequest POST request for a Geotab API call
+     *  Creates a GeotabApi call using Axios to differentiate between node and browser reqests.
+     *    Browser requests are done via XMLHTTPRequest
+     *    Node requests are done using http module
      *  @private
      *  @param {String} method The name of the API method
      *  @param {Object} params The method parameters
      *  @param {successCallback} callbackSuccess The function that is called if the method call was successful
      *  @param {failureCallback} [callbackError] The function that is called if the method call failed
-     *  @return {Object} An object with operations for the call. Supported operation(s): abort()
+     *  @return {Object} Promise object of Axios request. Allows for promise support
      */
-    callXHR = function (method, params, callbackSuccess, callbackError) {
-      var xhr = new XMLHttpRequest()
-      xhr.open('POST', getCallUrl(), true)
-      xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded')
-      xhr.addEventListener('abort', function (e) {
-        if (callbackError) {
-          callbackError('Cancelled', e)
-        }
-      })
-      xhr.onreadystatechange = function () {
-        if (xhr.readyState === 4) {
-          if (xhr.status === 200) {
-            var data,
-              error,
-              result
-            try {
-              data = JSON.parse(xhr.responseText)
-              if (data && data.error) {
-                error = data.error
-                debugLog(method, 'ERROR', error)
-                handleError(error, callbackError)
-              } else {
-                result = data.result
-                debugLog(method, 'SUCCESS', { result: result })
-                callbackSuccess(result)
-              }
-            } catch (e) {
-              handleError(e, callbackError)
-            }
-          } else {
-            debugLog(method, 'ERROR', xhr)
-            handleError(xhr, callbackError)
-          }
-        }
-      }
-      var rpcString
+    const callAxios = async function (method, params, callbackSuccess, callbackError) {
+      var rpcString;
       try {
         rpcString = JSON.stringify({
           method: method || '',
           params: params
-        })
+        });
       } catch (e) {
-        handleError(e, callbackError)
-        return
-      }
-      if (options.timeout) {
-        xhr.timeout = options.timeout * 1000
+        handleError(e, callbackError);
+        return; // Return ends the process
       }
-      xhr.send('JSON-RPC=' + encodeURIComponent(rpcString))
-      return {
-        abort: function () {
-          xhr.abort()
+
+      let config = {
+        headers: {
+          'Content-Type': 'applications/x-www-form-urlencoded'
```

Comment:
We could also consider moving this out of the API object. So it doesn't happen manically under the covers for you. Rather allow the constructor to accept paasword and sessionId. If session ID is provided, we will try to use it when making requests and not authenticate first. If session is not valid we will authenticate and advertise the new session so it can be saved by the application code.